### PR TITLE
common-cli: Improve command match

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: common-cli/src/main/java/co/cask/common/cli/CommandSet.java [hunk 1]
```diff
@@ -86,4 +76,15 @@ public CommandMatch findMatch(String input) throws InvalidCommandException {
...
+
+  /**
+   * Convert command pattern to regular expression that matches any input of this command
+   *
+   * @param pattern the command pattern
+   * @return regular expression
+   */
+  private String getMatchPattern(String pattern) {
+    String mandatoryPart = pattern.replaceAll("\\[.+\\]", ".*");
+    return mandatoryPart.replaceAll("<\\S+>", ".+");
```
Comment (jwang47) [inline common-cli/src/main/java/co/cask/common/cli/CommandSet.java]: Probably want to use `.+` instead of `\\S+`

**Comment by jwang47** [comment]
Could you also add some unit tests for this? I think it's for making commands with common prefixes work right?

**Comment by nitinmotgi** [comment]
Why are there no tests for the changes here. This is breaking the protocol that we have set forth for one that pre review needs to identify these issues and two how can one write code and not write tests. This to me is mind boggling.

**Comment by ghost** [comment]
Improve code and add test

**Comment by jwang47** [comment]
LGTM

**Pull request merged**